packages/abi-typegen/src/runTypegen.ts

```diff
@@ -52,10 +53,13 @@ export function runTypegen(params: IGenerateFilesParams) {
     Assembling file paths x contents
   */
   const abiFiles = filepaths.map((filepath) => {
+    const raw = readFileSync(filepath, 'utf-8');
+    const transpiled = JSON.stringify(transpileAbi(JSON.parse(raw)), null, 2);
     const abi: IFile = {
       path: filepath,
-      contents: readFileSync(filepath, 'utf-8'),
+      contents: transpiled,
     };
+
     return abi;
   });
```

Comment:
Might it be worth adding a try/catch around this operation?

Reply:
We could, but we never had it in place, and it always worked. It may be redundant and overly cautious since filepaths are extracted [from the disk](https://github.com/FuelLabs/fuels-ts/pull/2856/files/3b62ef47a32129ae8b6a9d8296435bb1109051c3#diff-1c28ec9246fc42304c8c01640e13f3a55f4fa8b7cb8672c36f936a4e08344da5R42) right before reading these files, so they will always exist.

Reply:
Completely understandable.